alpha/lib/model/categoryEntry.php

```diff
@@ -219,5 +219,120 @@ public function setCreatorPuserId ($v)
 	{
 		$this->putInCustomData('creatorPuserId', $v);
 	}
+
+	/**
+	 * @param $entryId
+	 * @param $categoryId
+	 * @throws KalturaErrors::CATEGORY_NOT_FOUND
+	 * @throws KalturaErrors::MAX_CATEGORIES_FOR_ENTRY_REACHED
+	 * @throws KalturaErrors::CANNOT_ASSIGN_ENTRY_TO_CATEGORY
+	 * @throws KalturaErrors::CATEGORY_ENTRY_ALREADY_EXISTS*
+	 * @throws KalturaErrors::INVALID_ENTRY_ID
+	 * @throws Exception
+	 * @return categoryEntry $categoryEntry
+	 */
+	public function add($entryId, $categoryId)
+	{
+		/*=========================================================*/
+		$entry = entryPeer::retrieveByPK($entryId);
+		if(!$entry)
+		{
+			throw new KalturaAPIException(KalturaErrors::INVALID_ENTRY_ID, $entryId);
+		}
+
+		$category = categoryPeer::retrieveByPK($categoryId);
```

Comment:
The entire function must be broken to smaller parts - 
function validateMaxCategoryEntries
function validateAccess 
function validateNotExist
function handleModeration
function assignUsers